sdk-extensions/autoconfigure/src/test/java/io/opentelemetry/sdk/autoconfigure/DefaultConfigPropertiesTest.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.autoconfigure;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+class DefaultConfigPropertiesTest {
+
+  @Test
+  void resourceAttributesMerged() {
+    Map<String, String> environment = new HashMap<>();
+    Map<String, String> system = new HashMap<>();
+
+    environment.put(
+        "OTEL_RESOURCE_ATTRIBUTES", "deployment.environment=env,service.name=env.service");
+    system.put("otel.resource.attributes", "deployment.environment=sys");
+    String expected =
+        "deployment.environment=env,service.name=env.service,deployment.environment=sys";
+
+    ConfigProperties props = DefaultConfigProperties.createForTest(system, environment);
+
+    assertThat(props.getString("otel.resource.attributes")).isEqualTo(expected);
```

Comment:
How does that work as a map though? Are the keys being properly merged?

Reply:
So the string in the config item ends up looking like `a=b,c=d,a=z' and when the map is built from this the last kv will win and non-repeated keys will not be overwritten.